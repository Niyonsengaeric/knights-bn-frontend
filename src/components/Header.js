import React from 'react';

const HeaderComponent = () => (
  <div>
    <h1 data-testid="header" className="box position__center">Barefoot Nomad</h1>
  </div>
);
export default HeaderComponent;
