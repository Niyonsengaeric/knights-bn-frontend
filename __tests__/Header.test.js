import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import Header from '../src/components/Header';

describe('===== Test Header =====', () => {
  afterEach(cleanup);
  it('renders without crushing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header />, div);
  });
  it('renders', () => {
    const { asFragment } = render(<Header text="Hello from Knights" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
