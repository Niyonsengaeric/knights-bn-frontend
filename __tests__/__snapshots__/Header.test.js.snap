// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`===== Test Header ===== renders 1`] = `
<DocumentFragment>
  <div>
    <h1
      class="box position__center"
      data-testid="header"
    >
      Barefoot Nomad
    </h1>
  </div>
</DocumentFragment>
`;
